src/test/regress/sql/citus_local_tables_mx.sql

```diff
@@ -381,5 +381,127 @@ SELECT run_command_on_workers(
 $$
 SELECT count(*) FROM pg_catalog.pg_tables WHERE tablename='citus_local_table_4'
 $$);
+
+-- test adding foreign table to metadata with the guc
+SET citus.use_citus_managed_tables TO ON;
+CREATE TABLE foreign_table_test (id integer NOT NULL, data text, a bigserial);
+INSERT INTO foreign_table_test VALUES (1, 'text_test');
+CREATE EXTENSION postgres_fdw;
+CREATE SERVER foreign_server
+        FOREIGN DATA WRAPPER postgres_fdw
+        OPTIONS (host 'localhost', port :'master_port', dbname 'regression');
+CREATE USER MAPPING FOR CURRENT_USER
+        SERVER foreign_server
+        OPTIONS (user 'postgres');
+CREATE FOREIGN TABLE foreign_table (
+        id integer NOT NULL,
+        data text,
+        a bigserial
+)
+        SERVER foreign_server
+        OPTIONS (schema_name 'citus_local_tables_mx', table_name 'foreign_table_test');
+
+--verify
+SELECT partmethod, repmodel FROM pg_dist_partition WHERE logicalrelid = 'foreign_table'::regclass;
+
+CREATE TABLE parent_for_foreign_tables (
```

Comment:
Querying `parent_for_foreign_tables` fails even if I try to setup things properly. Can you check how can we make this work?

```SQL
select * from parent_for_foreign_tables ;
NOTICE:  executing the command locally: SELECT project_id FROM "test sc1".parent_for_foreign_tables_102012 parent_for_foreign_tables
ERROR:  relation "test sc1.foreign_partition_1_102013" does not exist
CONTEXT:  remote SQL command: SELECT project_id FROM "test sc1".foreign_partition_1_102013
```

We seems to be sending the query to the shard, as far as I can see, the partitions should be created with `table_name` parameter:
```SQL
CREATE FOREIGN TABLE foreign_partition_1 PARTITION OF parent_for_foreign_tables FOR VALUES WITH (modulus 3, remainder 0) SERVER srv1 OPTIONS ( table_name 'foreign_partition_1');
```

Can you make sure that can execute insert and select queries on the parent and partition tables?

Reply:
I think to have a good user experience, we should aim to support updating tthe `OPTIONS` of the foreign tables:
```SQL

alter foreign table foreign_partition_2 OPTIONS (SET table_name  'foreign_partition_2');
ERROR:  alter table command is currently unsupported
DETAIL:  Only ADD|DROP COLUMN, SET|DROP NOT NULL, SET|DROP DEFAULT, ADD|DROP|VALIDATE CONSTRAINT, SET (), RESET (), ATTACH|DETACH PARTITION and TYPE subcommands are supported.
Time: 2.045 ms

lter foreign table foreign_partition_2 OPTIONS (ADD table_name  'foreign_partition_2');
ERROR:  alter table command is currently unsupported
DETAIL:  Only ADD|DROP COLUMN, SET|DROP NOT NULL, SET|DROP DEFAULT, ADD|DROP|VALIDATE CONSTRAINT, SET (), RESET (), ATTACH|DETACH PARTITION and TYPE subcommands are supported.

```

So that even if users forget to give the proper name, they can simply add or change it?

Reply:
It seems they are already supported, I missed that, see https://github.com/citusdata/citus/pull/5461/commits/017a6fdc3f9045bac5a02d21b5c0d8050edf97fa